src/backend/src/user_profile.rs

```diff
@@ -35,3 +40,31 @@ pub fn create_profile(
         default_profile
     }
 }
+
+pub fn add_credential(
+    principal: StoredPrincipal,
+    credential_type: &CredentialType,
+    user_profile_map: &mut StableBTreeMap<(u64, StoredPrincipal), Candid<StoredUserProfile>, VMem>,
+    user_profile_updated_map: &mut StableBTreeMap<StoredPrincipal, u64, VMem>,
+) -> Result<(), AddUserCredentialError> {
+    if let Ok(user_profile) = get_profile(principal, user_profile_map, user_profile_updated_map) {
```

Comment:
It would maybe be nice to scope the update of the user profile in an implementation? I mean not the persistence, just the contruction of the updated object.

Reply:
Good idea, done!